move buildinfo to `build/info.go` to simplify issue template

tbg opened the pull request:
The intention is to make reporting bugs easier by practically removing all of the build info steps.

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/6096)

<!-- Reviewable:end -->

tbg committed:
util: move build info to ./build

tschottdorf commented on ui/ts/models/proto.ts:
```diff
@@ -59,7 +59,7 @@ module Models {
-     * util/build.proto
+     * build/info.proto
      ****************************/
     export interface BuildInfo {
```
What regenerates this?

petermattis commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: 0 of 20 files reviewed at latest revision, 1 unresolved discussion.

---

*[ui/ts/models/proto.ts, line 64 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6096#-KFVlF80bDBl4EPf0-AY-r3-64:-KFZUmBXIc0NKmnhHXx6:1002020711) ([raw file](https://github.com/cockroachdb/cockroach/blob/7f7a876ee45bfbe25b2c542abc7baa136248ae88/ui/ts/models/proto.ts#L64)):*
Try `go generate` at the top-level.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/6096#-:-KFZUqh5gJwXYcew0CMc:1417088721)*
<!-- Sent from Reviewable.io -->

tschottdorf commented:
Review status: 0 of 20 files reviewed at latest revision, 1 unresolved discussion, all commit checks successful.

---

*[ui/ts/models/proto.ts, line 64 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6096#-KFVlF80bDBl4EPf0-AY-r3-64:-KFZh9wSmBf3rn9Yu9TF:-1373101580) ([raw file](https://github.com/cockroachdb/cockroach/blob/7f7a876ee45bfbe25b2c542abc7baa136248ae88/ui/ts/models/proto.ts#L64)):*
Did, but that didn't do it. Should be ok to merge like that, right?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/6096)*
<!-- Sent from Reviewable.io -->

bdarnell commented:
LGTM

---

Review status: 0 of 20 files reviewed at latest revision, 5 unresolved discussions, all commit checks successful.

---

*[.github/ISSUE_TEMPLATE.md, line 1 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6096#-KFZm_bwikWY-7yfjQrh:-KFZm_bwikWY-7yfjQri:1865428018) ([raw file](https://github.com/cockroachdb/cockroach/blob/7f7a876ee45bfbe25b2c542abc7baa136248ae88/.github/ISSUE_TEMPLATE.md#L1)):*
This file should be in a separate PR after we've made a release with the logging changes.

---

*[cli/start.go, line 160 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6096#-KFZnF6MuzOp-x3QUPvm:-KFZnF6MuzOp-x3QUPvn:272920905) ([raw file](https://github.com/cockroachdb/cockroach/blob/7f7a876ee45bfbe25b2c542abc7baa136248ae88/cli/start.go#L160)):*
Is this redundant with the logging in `clog.go`?

---

*[cli/start.go, line 195 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6096#-KFZnNNuysNoMgORQFFe:-KFZnNNuysNoMgORQFFf:514098346) ([raw file](https://github.com/cockroachdb/cockroach/blob/7f7a876ee45bfbe25b2c542abc7baa136248ae88/cli/start.go#L195)):*
We should use the same format here as in the logs.

---

*[ui/ts/models/proto.ts, line 64 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6096#-KFVlF80bDBl4EPf0-AY-r3-64:-KFZlboLhjwoUue2gA0G:-426631775) ([raw file](https://github.com/cockroachdb/cockroach/blob/7f7a876ee45bfbe25b2c542abc7baa136248ae88/ui/ts/models/proto.ts#L64)):*
Try `make -C ui`?

---

*[util/log/clog.go, line 946 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6096#-KFZlhvhgoOYaoVHCIJj:-KFZlhvhgoOYaoVHCIJk:-1819419712) ([raw file](https://github.com/cockroachdb/cockroach/blob/7f7a876ee45bfbe25b2c542abc7baa136248ae88/util/log/clog.go#L946)):*
This last line doesn't need to be in the grep used by the issue report. Also consider something like `config` instead of `log` for the marker used here.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/6096#-:-KFZnQq4DQDXL_289iD4:676834233)*
<!-- Sent from Reviewable.io -->

tamird commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 21 of 21 files at r1, 1 of 1 files at r2, 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 7 unresolved discussions, all commit checks successful.

---

*[cli/cli.go, line 25 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6096#-KF_UPhQa7922LKk1jXd:-KF_UPhQa7922LKk1jXe:-335821433) ([raw file](https://github.com/cockroachdb/cockroach/blob/a7a0894cec6c5bf7714e75186023ee58832ad51d/cli/cli.go#L25)):*
nit: should be below

---

*[cli/gen.go, line 24 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6096#-KF_UTTcGu79X5c8uIHv:-KF_UTTcGu79X5c8uIHw:-469765787) ([raw file](https://github.com/cockroachdb/cockroach/blob/a7a0894cec6c5bf7714e75186023ee58832ad51d/cli/gen.go#L24)):*
ditto

---

*[ui/ts/models/proto.ts, line 64 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6096#-KFVlF80bDBl4EPf0-AY-r3-64:-KF_Ufx6ampTjHyUWXXE:1113032875) ([raw file](https://github.com/cockroachdb/cockroach/blob/7f7a876ee45bfbe25b2c542abc7baa136248ae88/ui/ts/models/proto.ts#L64)):*
Nothing regenerates this, it's hand-written.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/6096#-:-KF_V2ReWEVDBQl3Tvpq:1417088721)*
<!-- Sent from Reviewable.io -->

tschottdorf commented:
Commit message updated for new log format. Will send the issue change as a separate PR.

---

Review status: 17 of 19 files reviewed at latest revision, 7 unresolved discussions.

---

*[cli/cli.go, line 25 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6096#-KF_UPhQa7922LKk1jXd:-KFcBhVEvGbWeCqgzlhX:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/a7a0894cec6c5bf7714e75186023ee58832ad51d/cli/cli.go#L25)):*
Done.

---

*[cli/gen.go, line 24 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6096#-KF_UTTcGu79X5c8uIHv:-KFcBjst704h2SaLs0pt:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/a7a0894cec6c5bf7714e75186023ee58832ad51d/cli/gen.go#L24)):*
Done.

---

*[cli/start.go, line 160 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6096#-KFZnF6MuzOp-x3QUPvm:-KFcC3G25lsEoCFsY1E2:757640429) ([raw file](https://github.com/cockroachdb/cockroach/blob/7f7a876ee45bfbe25b2c542abc7baa136248ae88/cli/start.go#L160)):*
Yeah, I think so. Removed.

---

*[cli/start.go, line 195 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6096#-KFZnNNuysNoMgORQFFe:-KFcHDnWngh-bGaN6XjI:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/7f7a876ee45bfbe25b2c542abc7baa136248ae88/cli/start.go#L195)):*
Done.

---

*[util/log/clog.go, line 946 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6096#-KFZlhvhgoOYaoVHCIJj:-KFcHCYnPJ3MIivJF7tA:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/7f7a876ee45bfbe25b2c542abc7baa136248ae88/util/log/clog.go#L946)):*
Done.

---

*[.github/ISSUE_TEMPLATE.md, line 1 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6096#-KFZm_bwikWY-7yfjQrh:-KFcBdFnynpfQUoJrv6-:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/7f7a876ee45bfbe25b2c542abc7baa136248ae88/.github/ISSUE_TEMPLATE.md#L1)):*
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/6096#-:-KFcHQ_JDR63AXUEhuGh:1046090474)*
<!-- Sent from Reviewable.io -->

tbg committed:
Groom log header

```
// omitting prefix `I160418 03:29:54.182165 util/log/clog.go:998`
[config] file created at: 2016/04/18 03:29:54
[config] running on machine: macts
[config] binary: CockroachDB beta-20160407-237-g3ed589e-dirty (darwin amd64, built 2016/04/18 07:28:45, go1.6)
[config] arguments: [./cockroach start]
line format: [IWEF]yymmdd hh:mm:ss.uuuuuu file:line msg
```

tamird commented:
Reviewed 2 of 2 files at r4, 2 of 2 files at r5, 1 of 1 files at r6.
Review status: all files reviewed at latest revision, 5 unresolved discussions, all commit checks successful.

---

*[cli/cli.go, line 25 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6096#-KF_UPhQa7922LKk1jXd:-KFd_6Grc1WS39L7XXF9:2090033692) ([raw file](https://github.com/cockroachdb/cockroach/blob/a7a0894cec6c5bf7714e75186023ee58832ad51d/cli/cli.go#L25)):*
doesn't look like it was done?

---

*[cli/gen.go, line 24 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6096#-KF_UTTcGu79X5c8uIHv:-KFd_1vjhCQbJrqHejSW:2090033692) ([raw file](https://github.com/cockroachdb/cockroach/blob/a7a0894cec6c5bf7714e75186023ee58832ad51d/cli/gen.go#L24)):*
doesn't look like it was done?

---

*[cli/start.go, line 195 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6096#-KFZnNNuysNoMgORQFFe:-KFd_9VvBELeSnkov2q7:2090033692) ([raw file](https://github.com/cockroachdb/cockroach/blob/7f7a876ee45bfbe25b2c542abc7baa136248ae88/cli/start.go#L195)):*
doesn't look like it was done?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/6096)*
<!-- Sent from Reviewable.io -->

tschottdorf commented:
Review status: all files reviewed at latest revision, 5 unresolved discussions, all commit checks successful.

---

*[cli/cli.go, line 25 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6096#-KF_UPhQa7922LKk1jXd:-KFezEiSsGPLKaL_k_PU:-974710764) ([raw file](https://github.com/cockroachdb/cockroach/blob/a7a0894cec6c5bf7714e75186023ee58832ad51d/cli/cli.go#L25)):*
It was done, but goimports re-rearranges them and I didn't notice.

---

*[cli/gen.go, line 24 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6096#-KF_UTTcGu79X5c8uIHv:-KFezIR9huj6nqrqNzR1:-974710764) ([raw file](https://github.com/cockroachdb/cockroach/blob/a7a0894cec6c5bf7714e75186023ee58832ad51d/cli/gen.go#L24)):*
It was done, but goimports re-rearranges them and I didn't notice.

---

*[cli/start.go, line 195 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6096#-KFZnNNuysNoMgORQFFe:-KFezt1xz_GC6nWFIj1-:-1164815401) ([raw file](https://github.com/cockroachdb/cockroach/blob/7f7a876ee45bfbe25b2c542abc7baa136248ae88/cli/start.go#L195)):*
Oh, I messed this one up. What I did change was to make the log header that is printed to the logs equal the standard log output format.

On this comment, honestly I like the old one better:

```
build:     beta-20160407-262-gcdd375a @ 2016/04/18 20:09:58 (go1.6)
admin:     http://localhost:8080
sql:       postgresql://root@localhost:26257?sslmode=disable
logs:      cockroach-data/logs
store[0]:  path=cockroach-data
```

vs

```
build:     CockroachDB beta-20160407-262-gcdd375a-dirty (darwin amd64, built 2016/04/18 20:12:50, go1.6)
admin:  http://localhost:8080
sql:       postgresql://root@localhost:26257?sslmode=disable
logs:      cockroach-data/logs
store[0]:  path=cockroach-data
```

that line is just pretty long and distracting.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/6096)*
<!-- Sent from Reviewable.io -->

tamird commented:
Review status: all files reviewed at latest revision, 5 unresolved discussions, all commit checks successful.

---

*[cli/cli.go, line 25 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6096#-KF_UPhQa7922LKk1jXd:-KFf0GqrepJrsshGsjM6:-1576821053) ([raw file](https://github.com/cockroachdb/cockroach/blob/a7a0894cec6c5bf7714e75186023ee58832ad51d/cli/cli.go#L25)):*
As you know, it doesn't rearrange if you leave a space.

---

*[cli/gen.go, line 24 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6096#-KF_UTTcGu79X5c8uIHv:-KFf0J3pNtWyaP3aGgqg:-1576821053) ([raw file](https://github.com/cockroachdb/cockroach/blob/a7a0894cec6c5bf7714e75186023ee58832ad51d/cli/gen.go#L24)):*
As you know, it doesn't rearrange if you leave a space.

---

*[cli/start.go, line 195 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6096#-KFZnNNuysNoMgORQFFe:-KFf0YU3CWcBQ_XMYjx3:1517754640) ([raw file](https://github.com/cockroachdb/cockroach/blob/7f7a876ee45bfbe25b2c542abc7baa136248ae88/cli/start.go#L195)):*
Still, why not use the same format in both place? I agree this one is nice and short. Add OS and platform to this and change the logged one to match.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/6096)*
<!-- Sent from Reviewable.io -->

tschottdorf commented:
Review status: all files reviewed at latest revision, 4 unresolved discussions, all commit checks successful.

---

*[cli/cli.go, line 25 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6096#-KF_UPhQa7922LKk1jXd:-KFjEfF_Vog6ZXwVs21N:297629984) ([raw file](https://github.com/cockroachdb/cockroach/blob/a7a0894cec6c5bf7714e75186023ee58832ad51d/cli/cli.go#L25)):*
My interactions with this list are usually limited to responding to your nits about the order (now that you mention it, it does make sense though) as I follow the religion of "if it's good enough for goimports, it's good enough for me".

---

*[cli/start.go, line 195 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6096#-KFZnNNuysNoMgORQFFe:-KFjEye2ggTXNeLFtjaL:-143599129) ([raw file](https://github.com/cockroachdb/cockroach/blob/7f7a876ee45bfbe25b2c542abc7baa136248ae88/cli/start.go#L195)):*
I think this is not worth a new PR but if you feel strongly, wanna take care of it? Good opportunity to fix the import order commented on above as well :-)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/6096)*
<!-- Sent from Reviewable.io -->

bdarnell commented:
Review status: all files reviewed at latest revision, 4 unresolved discussions, all commit checks successful.

---

*[cli/cli.go, line 25 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6096#-KF_UPhQa7922LKk1jXd:-KFjVgJ041ht3BAZCRlJ:395058932) ([raw file](https://github.com/cockroachdb/cockroach/blob/a7a0894cec6c5bf7714e75186023ee58832ad51d/cli/cli.go#L25)):*
Yes, if it's good enough for goimports, it's good enough. Tamir, please stop nitpicking about this.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/6096)*
<!-- Sent from Reviewable.io -->

mberhault commented on cli/start.go:
```diff
@@ -155,9 +156,6 @@ func runStart(_ *cobra.Command, _ []string) error {
 	}
 
-	info := util.GetBuildInfo()
-	log.Infof("[build] %s @ %s (%s)", info.Tag, info.Time, info.GoVersion)
```
it was kind of nice having the build info in stderr as well. Without it, you need to match stdout (which has no timestamp) with startup lines in stderr.

tschottdorf replied:
That makes sense, I'm not opposed to bringing it back.

On Thu, Apr 21, 2016 at 11:11 AM marc <notifications@github.com> wrote:

> In cli/start.go
> <https://github.com/cockroachdb/cockroach/pull/6096#discussion_r60598139>:
>
> > @@ -155,9 +156,6 @@ func runStart(_ *cobra.Command, _ []string) error {
> >  		return err
> >  	}
> >
> > -	info := util.GetBuildInfo()
> > -	log.Infof("[build] %s @ %s (%s)", info.Tag, info.Time, info.GoVersion)
>
> it was kind of nice having the build info in stderr as well. Without it,
> you need to match stdout (which has no timestamp) with startup lines in
> stderr.
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly or view it on GitHub
> <https://github.com/cockroachdb/cockroach/pull/6096/files/ef7138bc3fd7261ac40f84762c87f45846f41fe4#r60598139>
>
-- 

-- Tobias
